da/celestia/mock/server.go

```diff
@@ -70,137 +100,125 @@ func (s *Server) Stop() {
 		}
-		nShares = appendToShares(nShares, []byte{1, 2, 3, 4, 5, 6, 7, 8}, delimited)
-	}
-	shares := make([]Share, len(nShares))
-	for i := range nShares {
-		shares[i] = nShares[i].Share
-	}
-
-	resp, err := json.Marshal(namespacedSharesResponse{
-		Shares: shares,
-		Height: res.DAHeight,
-	})
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	s.writeResponse(w, resp)
-}
-
-func (s *Server) data(w http.ResponseWriter, r *http.Request) {
-	height, err := parseHeight(r)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	res := s.mock.RetrieveBlocks(r.Context(), height)
-	if res.Code != da.StatusSuccess {
-		s.writeError(w, errors.New(res.Message))
-		return
-	}
-
-	data := make([][]byte, len(res.Blocks))
-	for i := range res.Blocks {
-		data[i], err = res.Blocks[i].MarshalBinary()
+		s.writeResponse(w, bytes)
+	case "blob.GetAll":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
 		if err != nil {
 			s.writeError(w, err)
 			return
 		}
-	}
-
-	resp, err := json.Marshal(namespacedDataResponse{
-		Data:   data,
-		Height: res.DAHeight,
-	})
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
+		height := params[0].(float64)
+		block := s.mock.RetrieveBlocks(r.Context(), uint64(height))
+		blobs := make([]blob.Blob, len(block.Blocks))
+		for i, block := range block.Blocks {
+			data, err := block.MarshalBinary()
+			if err != nil {
+				s.writeError(w, err)
+				return
+			}
+			blobs[i].Data = data
 
-	s.writeResponse(w, resp)
-}
-
-func parseHeight(r *http.Request) (uint64, error) {
-	vars := mux2.Vars(r)
+		}
+		resp := &response{
+			Jsonrpc: "2.0",
+			Result:  blobs,
+			ID:      req.ID,
+			Error:   nil,
+		}
+		bytes, err := json.Marshal(resp)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		s.writeResponse(w, bytes)
+	case "share.SharesAvailable":
+		resp := &response{
+			Jsonrpc: "2.0",
+			ID:      req.ID,
+			Error:   nil,
+		}
+		bytes, err := json.Marshal(resp)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		s.writeResponse(w, bytes)
+	case "state.SubmitPayForBlob":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		if len(params) != 3 {
+			s.writeError(w, errors.New("expected 3 params"))
+			return
+		}
+		block := types.Block{}
+		blockBase64 := params[2].([]interface{})[0].(map[string]interface{})["data"].(string)
+		blockData, err := base64.StdEncoding.DecodeString(blockBase64)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		err = block.UnmarshalBinary(blockData)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
 
-	height, err := strconv.ParseUint(vars["height"], 10, 64)
-	if err != nil {
-		return 0, err
+		res := s.mock.SubmitBlock(r.Context(), &block)
+		resp := &response{
+			Jsonrpc: "2.0",
+			Result: &sdk.TxResponse{
+				Height: int64(res.DAHeight),
+			},
+			ID:    req.ID,
+			Error: nil,
+		}
+		bytes, err := json.Marshal(resp)
```

Comment:
this common code across the 4 cases can be pulled out to a local method:
```
bytes, err := json.Marshal(resp)
		if err != nil {
			s.writeError(w, err)
			return
		}
		s.writeResponse(w, bytes)
```
also, this is common across 3 cases:
```
var params []interface{}
		err := json.Unmarshal(req.Params, &params)
		if err != nil {
			s.writeError(w, err)
			return
		}
```